File: src/main/java/com/lkimilhol/healingwords/words/domain/Text.java
package com.lkimilhol.healingwords.words.domain;

import javax.persistence.Column;
import javax.persistence.Embeddable;

import lombok.NoArgsConstructor;

import com.lkimilhol.healingwords.exception.TextLengthExceedException;

@Embeddable
@NoArgsConstructor
public class Text {
    public static final int MAX_LENGTH = 255;

    @Column(name = "text", nullable = false)
    private String content;

    public Text(String content) {
        checkLength(content);
        this.content = content;
    }

    private void checkLength(String content) {
        if (content.length() > MAX_LENGTH) {
            throw new TextLengthExceedException();
        }
    }
}

File: src/main/kotlin/com/lkimilhol/healingwords/words/domain/Text.kt
package com.lkimilhol.healingwords.words.domain

import com.lkimilhol.healingwords.exception.TextLengthExceedException
import javax.persistence.Column
import javax.persistence.Embeddable

@Embeddable
class Text(
    @Column(name = "text", nullable = false)
    var content: String
) {

    fun hasForbidden(it: String): Boolean {
        return content.contains(it)
    }

    companion object {
        const val MAX_LENGTH = 255

        fun create(content: String): Text {
            checkLength(content)
            return Text(content)
        }

        private fun checkLength(content: String) {
            if (content.length > MAX_LENGTH) {
                throw TextLengthExceedException()
            }
        }
    }
}